### PR TITLE
Add gatk somatic tumour calling subworkflow

### LINKED ISSUE #1046: new subworkflow: gatk_tumour_only_somatic_variant_calling
This will add a subworkflow that will use the gatk modules to run tumour only variant calling, following a modified version of the gatk short somatic variant calling best practices: https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels-

The subworkflow will start with bam/cram file inputs and run mutect2 in tumour only mode, the calls from this will be passed  to filtermutectcalls, along with contamination and segmentation files generated by passing the inputs through getpileupsummaries and calculatecontamination (would it be worth making these two modules optional?).

The primary output will be the filtered vcf files, with the intention that they can be passed to another subworkflow, for annotation or joint variant calling depending on user needs.

This would help to close https://github.com/nf-core/sarek/issues/406 along with another subworkflow for paired tumour normal variant calling

- [x] This subworkflow does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this subworkflow
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this subworkflow
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by GCJMackenzie**
## PR checklist

Closes #1046 

This PR adds a subworkflow for performing somatic variant calling using gatk4 mutect in tumor only mode. The input files are run through getpileupsummaries and then calculatecontamination to produce the contamination and segmentation tables. These tables are passed along with the mutect2 calls into filtermutectcalls for filtering.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/main.nf [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        [meta, input_file[0], input_index[0]]
+    }
+    GETPILEUPSUMMARIES ( pileup_input , germline_resource , germline_resource_idx , interval_file )
+    ch_versions = ch_versions.mix(GETPILEUPSUMMARIES.out.versions)
+
+    //
+    //Contamination and segmentation tables created using calculatecontamination on the pileup summary table.
+    //
+    ch_pileup = GETPILEUPSUMMARIES.out.table.collect()
+    ch_pileup.add([])
```
Comment (lescai) [inline subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/main.nf:53]: understood this line adds an optional input needed for the module that's receiving this output.
not requesting a change, but would be nice to add a comment to the code to facilitate reading.

**Review by lescai — COMMENTED** [review]
Looks good to me. Docker is reported to have run inconsistently, without apparent reason. I would just recommend a double check and I'm happy to approve.

**Review by lescai — APPROVED** [review]
LGTM now

**Comment by maxulysse** [comment]
Same comments as https://github.com/nf-core/modules/pull/1067 for the `fastaidx` -> `fai` + `_idx` -> `_tbi`.
And as in my latest suggestion in #1067 I think we can use directly `input` as a channel instead of `ch_mutect2_in`.
But that's more open to discussion.

**Comment by maxulysse** [comment]
And of course, amazing work, I love it, I'm just super picky because it looks so good already :-)

**Comment by GCJMackenzie** [comment]
@maxulysse thanks for the input! changes should be implemented now, the actual input needs to be put into a channel so had to modify your suggestion on the paired PR, but other than that, all should be good.

File: subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/meta.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      description: list containing one BAM file indexe, also able to take CRAM index as an input
+      pattern: "[ *.{bam.bai/cram.crai} ]"
+  - fasta:
+      type: file
+      description: The reference fasta file
+      pattern: "*.fasta"
+  - fai:
+      type: file
+      description: Index of reference fasta file
+      pattern: "fasta.fai"
```
Comment (maxulysse) [inline subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/meta.yml:41]:
```suggestion
      pattern: "*.fasta.fai"
```

File: subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/meta.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      description: GATK sequence dictionary
+      pattern: "*.dict"
+  - germline_resource:
+      type: file
+      description: Population vcf of germline sequencing, containing allele fractions.
+      pattern: "*.vcf.gz"
+  - germline_resource_tbi:
+      type: file
+      description: Index file for the germline resource.
+      pattern: "*.vcf.gz_tbi"
```
Comment (maxulysse) [inline subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/meta.yml:53]:
```suggestion
      pattern: "*.vcf.gz.tbi"
```

File: subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/meta.yml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  - panel_of_normals:
+      type: file
+      description: vcf file to be used as a panel of normals.
+      pattern: "*.vcf.gz"
+  - panel_of_normals_tbi:
+      type: file
+      description: Index for the panel of normals.
+      pattern: "*.vcf.gz_tbi"
```
Comment (maxulysse) [inline subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/meta.yml:61]:
```suggestion
      pattern: "*.vcf.gz.tbi"
```

**Review by maxulysse — APPROVED** [review]

**Comment by maxulysse** [comment]
Sorry, I commited/pushed some tiny typo, just need to update the `tests/config/pytest_subworkflows.yml` file and we're good to go.

**Comment by GCJMackenzie** [comment]
@maxulysse @lescai That should be both this PR and PR #1067 ready to go I think. If you are both happy I will merge them both in.

@Emiller88 Are you happy for me to merge in just now?

**Review by Emiller88 — APPROVED** [review]

**Comment by Emiller88** [comment]
Looks great to me! Thanks for waiting for #933 to get merged

**Pull request merged**